algorithms/src/snark/marlin/ahp/prover/state.rs

```diff
@@ -114,13 +114,14 @@ impl<'a, F: PrimeField, MM: MarlinMode> State<'a, F, MM> {
 
                 let first_padded_public_inputs = &variable_assignments[0].0;
                 let input_domain = EvaluationDomain::new(first_padded_public_inputs.len()).unwrap();
-                let batch_size = variable_assignments.len();
+                let batch_size = variable_assignments.len().try_into()?;
                 total_instances += batch_size;
```

Comment:
these can overflow. we used `a.checked_add(b).ok_or_else(anyhow!("YOUR ERROR MSG"))?`, or alternative if you know with confidence that saturating is acceptable, then `a.saturating_add(b)`.

Reply:
it might be too much to do all of these in this PR :)

Reply:
Will scan and tackle such cases in a next PR!